src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -362,6 +450,130 @@ def launchCluster(self,
+        organized by availability zone.
+
+        The input subnet must be in the available state.
+
+        Similar subnets are ones with the same default-ness and ACLs.
+        """
+
+        # Grab the ec2 resource we need to make queries
+        ec2 = self.aws.resource(self._region, 'ec2')
+        # And the client
+        ec2_client = self.aws.client(self._region, 'ec2')
+
+        # What subnet are we basing this on?
+        base_subnet = ec2.Subnet(base_subnet_id)
+
+        # What VPC is it in?
+        vpc_id = base_subnet.vpc_id
+
+        # Is it default for its VPC?
+        is_default = base_subnet.default_for_az
+
+        # What ACLs does it have?
+        acls = set(self._get_subnet_acls(base_subnet_id))
+
+        # Compose a filter that selects the subnets we might want
+        filters = [{
+            'Name': 'vpc-id',
+            'Values': [vpc_id]
+        }, {
+            'Name': 'default-for-az',
+            'Values': ['true' if is_default else 'false']
+        }, {
+            'Name': 'state',
+            'Values': ['available']
+        }]
+
+        # Fill in this collection
+        by_az = {}
+
+        # Go get all the subnets. There's no way to page manually here so it
+        # must page automatically.
+        for subnet in self.aws.resource(self._region, 'ec2').subnets.filter(Filters=filters):
+            # For each subnet in the VPC
+
+            # See if it has the right ACLs
+            subnet_acls = set(self._get_subnet_acls(subnet.subnet_id))
+            if subnet_acls != acls:
+                # Reject this subnet because it has different ACLs
+                logger.debug('Subnet %s is a lot like subnet %s but has ACLs of %s instead of %s; skipping',
+                             subnet.subnet_id, base_subnet_id, subnet_acls, acls)
+                continue
+
+            if subnet.availability_zone not in by_az:
+                # Make sure we have a bucket of subnets for this AZ
+                by_az[subnet.availability_zone] = []
+            # Bucket the IDs by availability zone
+            by_az[subnet.availability_zone].append(subnet.subnet_id)
+
+        return by_az
+
+    @awsRetry
+    def _get_subnet_acls(self, subnet: str) -> List[str]:
+        """
+        Get all Network ACL IDs associated with a given subnet ID.
+        """
+
+        # Grab the connection we need to use for this operation.
+        ec2 = self.aws.client(self._region, 'ec2')
+
+        # Compose a filter that selects the default subnet in the AZ
+        filters = [{
+            'Name': 'association.subnet-id',
+            'Values': [subnet]
+        }]
+
+        # TODO: Can't we use the resource's network_acls.filter(Filters=)?
+        return [item['NetworkAclId'] for item in self._pager(ec2.describe_network_acls,
+                                                             'NetworkAcls',
+                                                             Filters=filters)]
```

Comment:
Is it possible for this to grow too large?  It's a habit now for me to make returning unbounded lists into generators that hand back one element (or a specified batch size) at a time for processing.

Reply:
I don't think people will put thousands of ACLs on a single subnet; there are only so many services in the world you could run, and each ACL is meant to hold multiple rules.

We could make this a generator, but how would that interact with retries? I don't think our default retry decorators would let us e.g. resume iteration from the right place.